lms/djangoapps/grades/tests/integration/test_events.py

```diff
@@ -174,11 +170,10 @@ def test_rescoring_events(self, models_tracker, handlers_tracker, instructor_tas
             choices=[False, False, False, True],
             choice_names=['choice_0', 'choice_1', 'choice_2', 'choice_3']
         )
-        module_store = modulestore()
-        with module_store.branch_setting(ModuleStoreEnum.Branch.draft_preferred, self.course.id):
+        with self.store.branch_setting(ModuleStoreEnum.Branch.draft_preferred, self.course.id):
             self.problem.data = new_problem_xml
-            module_store.update_item(self.problem, self.instructor.id)
-            module_store.publish(self.problem.location, self.instructor.id)
+            self.store.update_item(self.problem, self.instructor.id)
+        self.store.publish(self.problem.location, self.instructor.id)
```

Comment:
**Note to reviewers:** This `publish` call wasn't updating the cached Block Structure.  Moving the call out of the `branch_setting` context-manager fixed the issue.